This reverts ASTERISK-23818 as the fix causes lua modules to break.
Note sent upstrean and hopefully they fix it in more proper way.

diff -ru asterisk-12.4.0.orig/pbx/pbx_lua.c asterisk-12.4.0/pbx/pbx_lua.c
--- asterisk-12.4.0.orig/pbx/pbx_lua.c	2014-08-04 15:11:22.000000000 -0300
+++ asterisk-12.4.0/pbx/pbx_lua.c	2014-08-05 10:25:35.092623642 -0300
@@ -1672,7 +1672,7 @@
 	return AST_MODULE_LOAD_SUCCESS;
 }
 
-AST_MODULE_INFO(ASTERISK_GPL_KEY, AST_MODFLAG_DEFAULT, "Lua PBX Switch",
+AST_MODULE_INFO(ASTERISK_GPL_KEY, AST_MODFLAG_GLOBAL_SYMBOLS, "Lua PBX Switch",
 		.load = load_module,
 		.unload = unload_module,
 		.reload = reload,
